java/integTest/src/test/java/glide/ConnectionTests.java

```diff
@@ -202,4 +243,74 @@ public void test_az_affinity_non_existing_az() {
         assertEquals(4, matchingEntries);
         azTestClient.close();
     }
+
+    @SneakyThrows
+    @ParameterizedTest
+    @ValueSource(booleans = {true, false})
+    public void test_connection_timeout(boolean clusterMode) {
+        var backoffStrategy =
+                BackoffStrategy.builder().exponentBase(2).factor(100).numOfRetries(1).build();
+        var client = createConnectionTimeoutClient(clusterMode, 250, 20000, backoffStrategy);
+
+        // Run a long-running DEBUG SLEEP command to simulate a blocking operation
+        CompletableFuture<Void> runDebugSleep =
+                CompletableFuture.runAsync(
+                        () -> {
```

Comment:
custom command already returns a future, no need to wrap it

Reply:
not sure I understand, then how should this look like?

Reply:
`var f = client.custom([...])` without `.get()`